src/transaction/undo_buffer.cpp

```diff
@@ -134,6 +134,9 @@ UndoBufferProperties UndoBuffer::GetProperties() {
 			if (info->is_consecutive) {
 				properties.estimated_size += sizeof(row_t) * info->count;
 			}
+			if (info->table->HasIndexes()) {
+				properties.has_index_deletes = true;
+			}
```

Comment:
Should we also add a test for non-unique indexes? Also, seems more broad, i.e., not only for deletes/changes to indexes? I don't think it makes a significant difference, just wondering if I understand correctly.